### DISCUSSION TITLE
Como você define um dia produtivo?

### DISCUSSION THREAD

**Discussion by felquis** [frontendbr/forum#266] [Discussão]
Estou sendo produtivo ou não, thats the question! 

Pessoalmente tenho dificuldade de definir se meu dia é produtivo ou não, e tento alinhar com gestores o que é algo produtivo, as vezes sinto que não estou produtivo, porém ao conversar com alguém descubro que na realidade estava sendo produtivo. 

Com este tópico, gostaria de descobrir o que vocês usam como parâmetros para decidir que seu dia foi produtivo o suficiente e descansar sossegado. 

Seus gestores ajudam a alinhar as expectativas de o que é ser produtivo? Se sim, como o fazem? 

Usam ferramentas que ajudam a decidir se o trabalho está sendo produtivo? Se sim qual e como usam? 

Muito importante também separar se o seu contexto de trabalho é em produto, agência, freela ou estudo. 

Acredito ser uma discussão válida

Att.

**Comment by woliveiras** [comment]
Excelente tópico cara!

**Comment by raulfernando** [comment]
Muito bom o tópico, eu geralmente antes de começar a trabalhar defino umas metas e faço o possível para finalizadas os mais rapido possivel

**Comment by barbier** [comment]
IMO:
A sprint foi definida e a equipe alinhou a ordem das histórias que devem ser feitas.
O dev está fazendo as histórias na estimativa de tempo definida na pontuação? Então sim.

**Comment by Cleysonlb** [comment]
Cara pra medir isso é legal você trabalhar em cima de metas pré estabelecidas, tipo, organiza um conjunto de coisas para fazer no dia, e tenta ir resolvendo, definindo um tempo médio para cada tarefa e tals, claro que não da pra ser 100% produtivo todo dia, mas é interessante pra medir tarefa x tempo

**Comment by vitocaio** [comment]
Ótimo tópico. 
Acredito que isso varia muito de acordo com o que é produtividade para você, se esta falando em entregas(issues) ou etapas de um determinado trabalho, gosto de mensurar minhas atividades dar pontos para elas de acordo com a dificuldade igual ao scrum e então entregar o máximo de pontos possível, se bem que ultimamente não estou conseguindo me organizar para fazer isto.

**Comment by rubenmarcus** [comment]
www.rescuetime.com , uso ele como ferramenta pra medir produtividade, alem das tasks do Asana..

**Comment by victor-vct** [comment]
Tópico show mesmo! 
Quando estava estudando, no caso dev Android, por conta, eu gostava de colocar metas por tema...então tinha até X data para aprender tal capítulo. Quando terminava gostava de analisar o quanto eu tinha aprendido, o quanto dominava o que estava no capitulo. Se ainda tinha tempo, e não estava dominando, revia o capitulo, se tinha dado o tempo eu revia pra ver se errei no jeito de estudar aquele assunto.
Agora no trabalho ainda não consegui uma forma efetiva de definir quão produtivo estou sendo. Alinhamos o que deve ser entregue e faço, se consigo terminar antes tento ver melhorias mas quando não consigo fico pensando se errei na estrategia.

**Comment by rfviolato** [comment]
Já trampei em empresas com sprints muito bem estruturados e tal e de fato um sprint bem dividido e que não é irrealista é muito bom pra você e e o time que você trampa sentirem-se produtivos.

Porém como estou atualmente numa empresa que organiza bem pouco os sprints (por diversos fatos como equipes de um homen só) eu particularmente gosto de, ao final do dia, fazer uma pequena reflexão: Tento lembrar de como estava o progresso antes de eu começar a codar no dia e o que construi ao final deste mesmo dia. Abro o histórico de commits do repo que commitei no dia, por exemplo, e dou uma olhada neles para me ajudar a lembrar tudo que fiz. Gosto de fazer este exercício e ele me ajuda muito a sentir-me produtivo (ou não, rs).

**Comment by woliveiras** [comment]
Acho que estipular metas logo no começo do dia é o melhor caminho, pois da pra mensurar nossa produtividade individualmente.

Você pega o que tem pra fazer, quebra em pequenas tarefas e vai resolvendo.

Acho que se colocarmos como `finalizar issues` como produtividade, o dia em que ajudarmos alguém a resolver um problema que demore um tempo ou acontecerem muitas reuniões estaremos ficando como improdutivos, mas isso não seria produtividade (tirando as reuniões chatas, aaaaaaff)?

Fora que se focarmos em fechar issues, podemos estar sendo produtivos a curto prazo, mas matando nossa produtividade no futuro, pois podemos acabar focando em quantidade ao invés de qualidade.

Também pode acontecer que você esteja trabalhando numa história, ai da um B.O. e você precisa resolver, o Gestor pode pedir que você atue em outras coisas... Vish! pode acontecer muita coisa.

Com essas pequenas metas você já consegue se sentir produtivo.

Já pra mostrar essa produtividade pro Gestor/Liderança seria coletar números números, tipo: finalizei tantas histórias, ajudei outra pessoa a finalizar essa outra história, entreguei determinada feature completa, criei componente X que vai ajudar em futuras entregas.

Pra mostrar valor precisa ter certeza do que expor. A melhor maneira de se fazer isso é tirando métricas de como as coisas estão hoje e como ficou depois das entregas e quanto tempo demorou para entregar.

**Comment by fernandofleury** [comment]
pra mim é produtivo quando eu consigo avançar nos tópicos que julguei essenciais pra aquele dia :)

não gosto muito de complexar essa questão

**Comment by felipefialho** [comment]
Sou uma pessoa bem metódica e apegado a rotinas.
- Geralmente tenho um check-list mental de todas as tarefas que preciso terminar no dia, pessoais e profissionais.
- Além disso, vou colocando tarefas pessoais e profissionais no Google Agenda para os próximos dias (ir no médico, escrever artigo, etc..), assim não esqueço.
- No trabalho usamos o JIRA para definir as issues importantes para serem feitas no dia.

Sendo assim, considero um dia produtivo quando consigo realizar as tarefas do meu check-list mental, as que estavam no Google Agendas e no JIRA.

E na verdade fico bastante incomodado quando não consigo concluir alguma coisa.

**Comment by ecarrara-araujo** [comment]
Correndo o risco de ser um pouco repetitivo, mas vamos lá.
Produtividade é um termo que, na minha opinião, causa muita confusão pois as pessoas tem visões diferentes sobre o assunto e é muito dificil chegar a um consenso.
Pessoalmente gosto de abordar o assunto de forma simples: eu consegui finalizar aquilo que me propus no começo do dia? Se sim ótimo, senão por que?

A nossa vida tem vários aspectos que influenciam a nossa taxa de execução de tarefas, temos que tomar cuidado para não ser injustos com nós mesmos quando medimos nossa "produtividade".

Na prática lido com isto preparando uma lista de manhã, ou na noite do dia anterior, de tudo que preciso resolver naquele dia (incluindo reuniões e outros eventos), depois tento estimar se consigo resolver tudo aquilo com base no tempo que tenho disponível naquele dia, sem esquecer de considerar que temos almoço, janta, trânsito, café e por ai vai. Nessa lista incluo coisas de todos os aspectos da minha vida pessoal e profissional. Isso me ajuda muito, durante o dia consigo saber o que devo fazer e ao final do dia posso dizer que o dia foi produtivo se eu resolvi minhas tarefas, e se não as resolvi consigo replanejar ou me adaptar.

Quando temos que lidar com chefes, gerentes, clientes e etc é muito importante chegar a um consenso sobre o que é ser produtivo e isso varia muito de situação para situação, conversar abertamente é fundamental para se chegar a métricas formais ou informais de produtividade que sejam relevantes para o projeto que você está trabalhando e para sua carreira.

Para quem gosta de métodos eu tenho usado o [Bullet Journaling](http://www.thelazygeniuscollective.com/blog/how-to-bullet-journal) e tem sido fantástico!

**Comment by leandroruel** [comment]
quando durmo mal meu dia é 99,9% improdutivo. não tem como.

**Comment by yanmagale** [comment]
Cara ... excelente tópico. Realmente é um assunto que todo mundo tem algum tipo de dúvida.

Concordo muito com as falas do @woliveiras e do @ecarrara-araujo . Uma coisa que eu aprendi, é que infelizmente não se pode ser produtivo em todos os momentos, no sentido que somos seres humanos. Vão existir dias que você não vai conseguir fazer a feature da forma mais rápida, então nestes dias não adianta forçar o cérebro porque você não vai conseguir avançar. Como também existem aqueles dias que a "alma de programador" bate em sua cabeça e você sai desenvolvendo como nunca. Nestes  dias também é bacana você tentar fazer um pouco além, para compensar os dias que você não estava tão produtivo. Dessa forma eu consigo equilibrar as coisas e saber se eu tive uma sprint produtiva ou não.

Diariamente, procuro anotar as tarefas no [Workflowy](https://workflowy.com) ,  pessoais e profissionais, e ao final do dia eu faço um review. Especificamente no trabalho, eu gosto de avaliar a história que estou trabalhando: Se eu estou trabalhando numa história que eu consigo acabá-la em um dia(geralmente features simples, ajustes e alterações.), a meta é finalizá-la. Caso sobre algum tempo, faço alguns refactorings, senão, tento fazer isso logo no começo do dia seguinte. Já para histórias maiores(que demandariam 2, 3 ou 4 dias de trabalho) o que eu faço é tentar quebrar a tarefa em partes, de forma que eu consiga fazer a maneira mais funcional mais rapidamente. Nisso eu consigo mensurar o quanto eu evolui na tarefa.

Acho que isso também é muito de cada um e principalmente no ambiente em que se está.

**Comment by santosluiz** [comment]
Excelente tópico. Ter um dia produtivo ou não produtivo vai muito além de cumprir o objetivo. Pois, pode haver casos em que você não cumpriu o objetivo mas descobriu uma nova ferramenta ou até mesmo um modo de resolver algum outro problema que você e/ou sua equipe está passando. É relativo. Muito relativo.
Entretanto, de forma geral,, eu creio que ter um dia produtivo é quando cumpro minhas metas estabelecidas. Seja uma meta alocada na minha mente ou escrita num post-it da vida. Além disso, eu também considero um dia produtivo quando aprendo algo novo! Desde uma nova tag HTML5 ou um funcionamento de determinada função. Qualquer coisa!
